src/test/ui/issues/issue-20605.stderr

```diff
@@ -2,10 +2,10 @@ error[E0277]: the size for values of type `dyn Iterator<Item = &'a mut u8>` cann
   --> $DIR/issue-20605.rs:2:17
    |
 LL |     for item in *things { *item = 0 }
-   |                 ^^^^^^^ expected an implementor of trait `IntoIterator`
+   |                 ^^^^^^^ expected an implementor of trait `iter::loop_desugar::IntoIterator`
```

Comment:
This feels like a pretty unfortunate diagnostics regression.

Reply:
I named the trait and method the way they are so that the diagnostics look at least mostly right. If I rename them then this would get worse.